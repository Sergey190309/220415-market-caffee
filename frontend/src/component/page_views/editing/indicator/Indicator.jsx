import React from 'react'
import PropTypes from 'prop-types'

const Indicator = () => {
  /**
   * Component shows name of the componet could be edited on right click.
   */
  return (
    <div>Indicator</div>
  )
}

Indicator.defaultProps = {}
Indicator.propTypes = {}

export default Indicator