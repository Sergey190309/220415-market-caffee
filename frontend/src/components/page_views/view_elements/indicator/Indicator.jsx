import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Popup } from 'semantic-ui-react'

const Indicator = ({ isOpened, context, header, content }) => {
  const [opened, setOpened] = useState(isOpened)
  // useEffect(() => {
  //   let mounted = true
  //   if (mounted === true) {
  //     setOpened(isOpened)
  //   }
  //   return () => {
  //     mounted = false
  //   }
  // }, [isOpened])

  // useEffect(() => {
  //   setOpened(isOpened)
  // }, [isOpened])
  // console.log('components, page_view, view_elements, indicator, header ->', header)
  // console.log('components, page_view, view_elements, indicator, content ->', content)
  return (
    <Popup
      // disabled={false}
      position='right center'
      context={context}
      open={opened}
      header={header}
      content={content}
      onClose={() => {
        setOpened(false)
        // setIndicatorOpened(false)
      }}
    />
  )
}

Indicator.defaultProps = {
  isOpened: false,
  context: {},
  header: '',
  content: ''
}

Indicator.propTypes = {
  isOpened: PropTypes.bool.isRequired,
  context: PropTypes.object.isRequired,
  header: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
}
export default Indicator
