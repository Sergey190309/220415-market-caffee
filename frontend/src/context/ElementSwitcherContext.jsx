import React from 'react'

export const ElementSwitcherContext = React.createContext()

export const ElementSwitcherProvider = ElementSwitcherContext.Provider
export const ElementSwitcherConsumer = ElementSwitcherContext.Consumer
