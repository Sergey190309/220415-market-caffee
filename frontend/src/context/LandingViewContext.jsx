import React from 'react'

const LandingContext = React.createContext()
// export const LandingContext = React.createContext()

export const LandingProvider = LandingContext.Provider
export const LandingConsumer = LandingContext.Consumer
