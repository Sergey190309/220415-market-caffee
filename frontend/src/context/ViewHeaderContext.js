// import React from 'react'

// export const ViewHeaderContext = React.createContext()

// export const ViewHeaderProvider = ViewHeaderContext.Provider
// export const ViewHeaderConsumer = ViewHeaderContext.Consumer
