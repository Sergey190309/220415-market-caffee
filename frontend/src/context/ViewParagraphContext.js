// import React from 'react'

// export const ViewParagraphContext = React.createContext()

// export const ViewParagraphProvider = ViewParagraphContext.Provider
// export const ViewParagraphConsumer = ViewParagraphContext.Consumer
