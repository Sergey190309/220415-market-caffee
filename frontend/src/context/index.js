export {
  LandingContext,
  LandingProvider,
  LandingConsumer
} from './LandingViewContext'

export {
  ElementSwitcherContext,
  ElementSwitcherProvider,
  ElementSwitcherConsumer
} from './ElementSwitcherContext'

// export {
//   ViewVBlockContext,
//   ViewVBlockProvider,
//   ViewVBlockConsumer
// } from './ViewVBlockContext'

// export {
//   ViewParagraphContext,
//   ViewParagraphProvider,
//   ViewParagraphConsumer
// } from './ViewParagraphContext'
