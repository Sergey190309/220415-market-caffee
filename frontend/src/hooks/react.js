import {useState, useEffect, useRef, useCallback } from 'react'

export const useAppState = useState
export const useAppEffect = useEffect
export const useAppRef = useRef
export const useAppCallback = useCallback
