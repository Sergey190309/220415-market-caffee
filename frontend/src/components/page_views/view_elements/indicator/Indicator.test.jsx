import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import Indicator from './Indicator'
import userEvent, { specialChars } from '@testing-library/user-event'
import { LandingProvider } from '../../../../context'

describe('Indicator', () => {
  const testProps = {
    isOpened: false,
    context: {},
    content: 'mockContent',
    setIndicatorOpened: jest.fn()
  }
  const mockContext = {
    componentName: { componentName: 'landing' }
  }
  const contextRender = (actualProps, context = mockContext) => {
    render(
      <LandingProvider value={context.componentName} >
        <Indicator {...actualProps} />
      </LandingProvider>
    )
  }
  describe('Appeance', () => {
    test('indicator exists (snapshot)', async () => {
      const actualProps = { ...testProps, isOpened: true }
      await waitFor(() => {
        contextRender(actualProps)
      })
      const popup = screen.getByTestId('Popup')
      expect(popup.children).toHaveLength(2)
      expect(popup.children[0])
        .toHaveTextContent(mockContext.componentName.componentName)
      expect(popup.children[1])
        .toHaveTextContent(testProps.content)
      expect(popup).toMatchSnapshot()
      // screen.debug(popup)
    })
    test('inidcator does not exitst', async () => {
      const actualProps = { ...testProps }
      await waitFor(() => {
        contextRender(actualProps)
      })
      // const popup = screen.getByTestId('Popup')
      expect(screen.queryByTestId('Popup'))
        .not.toBeInTheDocument()
    })
  })
  describe.skip('Actions', () => {
    /**
     * I'm unable to simulate esc and click out of element.
     */
    test('escape', async () => {
      const actualProps = {
        ...testProps,
        isOpened: true
      }
      await waitFor(() => {
        contextRender(actualProps)
      })
      // const popup = screen.getByTestId('Popup')
      expect(screen.getByTestId('Popup'))
        .toBeInTheDocument()
      userEvent.keyboard(specialChars.escape)
      // await waitFor(() => {
      // })
      console.log('Indicator.test:\n escape',
        '\n  specialChars.escape ->', specialChars.escape)
      screen.debug(screen.getByTestId('Popup'))
      // expect(screen.queryByTestId('Popup'))
      //   .not.toBeInTheDocument()
    })
    test('click outside', async () => {
      const actualProps = { ...testProps, isOpened: true }
      await waitFor(() => {
        contextRender(actualProps)
      })
      // const popup = screen.getByTestId('Popup')
      expect(screen.queryByTestId('Popup')).not.toBe(null)
      userEvent.click(document.body)
      // expect(screen.queryByTestId('Popup')).toBe(null)
      screen.debug(document.body)
    })
  })
})
