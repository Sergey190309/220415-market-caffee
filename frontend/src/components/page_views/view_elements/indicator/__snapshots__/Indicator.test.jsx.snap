// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator Appeance indicator exists (snapshot) 1`] = `
<div
  class="ui top left popup transition visible"
  data-testid="Popup"
  style="position: initial;"
>
  <div
    class="header"
  >
    landing
  </div>
  <div
    class="content"
  >
    mockContent
  </div>
</div>
`;
